streams/test-utils/src/main/java/org/apache/kafka/streams/processor/MockProcessorContext.java

```diff
@@ -319,7 +321,7 @@ public void setRecordMetadata(final String topic,
         this.topic = topic;
         this.partition = partition;
         this.offset = offset;
-        this.headers = headers;
+        this.headers = Objects.requireNonNull(headers);
```

Comment:
This change is a little tricky... But if we consider it a bug-fix, it might still be ok?

Reply:
If you want to introduce the invariant without introducing an NPE regression, you could just coerce a null:

```suggestion
        this.headers = headers == null ? new RecordHeaders() : headers;
```

Reply:
Thinking about is more, it might be best to keep the code as is, and allow `null`.
  - adding the new check might break existing test code
  - converting `null` to empty header might also break tests
  - users don't have a good way to pass in headers, as `Headers` is an interface (and there is no other public API they could use to create an object)
  - it was not a problem so far, and long term we want to deprecate this "old" API anyway (so we might want to avoid a case of pre-mature optimization)